docs/workshops/interviews/technical/workshop.md

```diff
@@ -0,0 +1,262 @@
+> `"I".length` is 1 which is not greater than `"I".length"` which is also 1 so we skip the body of the `if`, and move on to the next word.
+>
+> `"like".length` is 4 which is greater than 1, so we set `longestWord` to `"like"`, and continue around the loop.
+>
+> `"cats".length` is 4, which is not greater than 4, so we skip the `if`.
+>
+> That's the end of the loop, so we will return `"like"`, which is the longest word of the sentence.
+
+If you find a problem with your code, say, and fix it! We all make bugs - by noticing the bug yourself, you're showing the interviewer that you pay attention to your code and whether it's right.
+
+It's a good idea to talk through each of the edge-cases you came up with earlier in the interview.
+
+If you're writing code on a text editor or whiteboard, it can also be useful to write values of variables next to them while you're talking through the code, so you don't have to remember them (and to make it easier for your interviewer to follow) - you can delete/erase the old values when they get overwritten. So the above code, in the middle of talking through an example, may look like:
+
+```js
+function longestWord(sentence) {
+    const sentenceWithoutPunctuation = sentence.replace(/[^A-Za-z ]g/, ""); // "I like cats"
+    const words = sentenceWithoutPunctuation.split(" "); // ["I", "like", "cats"]
+    let longestWord = words[0]; // "I"
+    for (const word of words) { // word = "like"
+        if (word.length > longestWord.length) { // 4 > 1
+            longestWord = word;
+        }
+    }
+    return longestWord;
+}
+```
+
+#### Suggest possible refactorings/improvements
+
+If you think you'd like to change your code to improve it (e.g. to improve a variable name, or to extract a function for a piece of repeated code), describe how you would like to change it to your interviewer.
+
+Because writing code on whiteboards or shared text environments can be fiddly, the interviewer may just say "That's a good idea, but let's not bother", or they may ask you to make the change. But _you_ suggesting an improvement highlights that you care about your code quality.
+
+Of course, if you think your code is fine as it is, that's fine too!
+
+## Answering questions about your solution
+
+Your interviewer may now ask you some questions about your solution.
+
+Some examples of things the interviewer may ask about:
+* Why you chose to do something one way rather than another (e.g. why you used a `for` loop instead of a call to `map`).
+* Whether there are any interesting edge cases your code doesn't handle.
+* If there is anything you would change about the code.
+* TODO: Think about how to discuss: If there are any trade-offs you've made in your code
+* TODO: Think about whether to discuss efficiency at all.
+* TODO: Expand this list.
+
+:::tip
+Asking you about other ways of solving the problem does not mean you did it wrong! It can still be interesting to talk about other ways to do things, even if yours is a good one!
```

Comment:
I wonder if it's worth mentioning that the interviewer has probably seen lots of solutions to the problem, and therefore usually has a much better understanding than you do when seeing it for the first time.